scheduling-manager/src/test/java/com/amazonaws/blox/scheduling/state/ContainerInstanceListerTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling.state;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import org.mockito.Mockito;
+import software.amazon.awssdk.services.ecs.ECSAsyncClient;
+import software.amazon.awssdk.services.ecs.model.ContainerInstance;
+import software.amazon.awssdk.services.ecs.model.DescribeContainerInstancesRequest;
+import software.amazon.awssdk.services.ecs.model.DescribeContainerInstancesResponse;
+import software.amazon.awssdk.services.ecs.model.ListContainerInstancesRequest;
+import software.amazon.awssdk.services.ecs.model.ListContainerInstancesResponse;
+
+public class ContainerInstanceListerTest {
+
+  private static final String CLUSTER_1 = "cluster1";
+
+  // TODO: Add tests for synchronous/asynchronous failures
+
+  @Test
+  public void returnsEmptyListWhenNoContainerInstances() {
+    ECSAsyncClient ecs = mock(ECSAsyncClient.class);
+    when(ecs.listContainerInstances(any()))
+        .thenReturn(
+            CompletableFuture.completedFuture(
+                ListContainerInstancesResponse.builder().containerInstanceArns().build()));
+
+    ContainerInstanceLister containerInstances = new ContainerInstanceLister(ecs, CLUSTER_1);
+
+    List<ClusterSnapshot.ContainerInstance> describe = containerInstances.describe().join();
+
+    assertThat(describe, empty());
+  }
+
+  @Test
+  public void returnsContainerInstancesFromAllPages() {
+    ECSAsyncClient ecs = mock(FakeECSAsyncClient.class, Mockito.CALLS_REAL_METHODS);
```

Comment:
is this one of the temporary unit/integ mix tests? can we move them to a separate package so they're easy to refactor into integ tests later?

Reply:
No, this unit test has value, and will need to be expanded to cover failure cases that are hard/impossible to reproduce against a real ECS client (e.g. partial asynchronous failure)

Reply:
Oh, and to answer the second part of your question: They are pure unit tests, since they mock out all dependencies, do I don't think they belong in a separate package.